src/Features/LanguageServer/Protocol/WellKnownLspServerKinds.cs

```diff
@@ -8,6 +8,11 @@ namespace Microsoft.CodeAnalysis.LanguageServer;
 
 internal enum WellKnownLspServerKinds
 {
+    /// <summary>
+    /// Razor LSP server for Razor document requests (.razor and .cshtml files)
+    /// </summary>
+    RazorCohostServer,
```

Comment:
Item 2 hadn't really occurred to me, but I guess we could do that. Without dynamic registration though, we probably can't do that behind a feature flag though, as it would just be controlled by the content type attribute on the server. Interesting idea though.

The rest is correct though. Essentially this lets us work on migration, find bugs, maybe get some benefit, but without being blocked on dynamic registration.